odrive-motor/src/odriveCAN.py

```diff
@@ -114,34 +113,35 @@ async def reconfigure(self, config: ComponentConfig, dependencies: Mapping[Resou
-        await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+        self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
         await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
-        await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x01, 'Input_Mode': 0x01})
-        await self.send_can_message('Set_Input_Torque', {'Input_Torque': torque})
+        self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x01, 'Input_Mode': 0x01})
+        self.send_can_message('Set_Input_Torque', {'Input_Torque': torque})
 
     async def go_for(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
         rps = rpm / MINUTE_TO_SECOND
 
         if revolutions == 0.0:
-            await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x02, 'Input_Mode': 0x01})
-            await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+            self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x02, 'Input_Mode': 0x01})
+            self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
             await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
-            await self.send_can_message('Set_Input_Vel', {'Input_Vel': rps, 'Input_Torque_FF': 0})
+            self.send_can_message('Set_Input_Vel', {'Input_Vel': rps, 'Input_Torque_FF': 0})
 
         else:
-            await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
-            await self.send_can_message('Set_Limits', {'Velocity_Limit': 10.0, 'Current_Limit': 40.0})
-            await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+            self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
+            self.send_can_message('Set_Traj_Vel_Limit', {'Traj_Vel_Limit': rps})
+            self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
             await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
 
             current_position = await self.get_position()
             if rpm > 0.0:
                 goal_position = current_position+revolutions+self.offset
-                await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
+                self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
             else:
                 goal_position = current_position-revolutions+self.offset
-                await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
-            await self.wait_and_set_to_idle(goal_position)
+                self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
+            
+            self.wait_and_set_to_idle(goal_position)
```

Comment:
There are two possible behaviors and I'm not sure which one is better. If I `await` this function, on app the motor will say idle during the entire rotation and not update position, but once the command is finished it will set the control state back to idle and update the position accordingly. If I leave it as is, the card on app says running and updates the position as it's spinning, but it doesn't set the control state back to idle once it's done. The serial model is currently implemented with the second option of expected frontend behavior and not reseting to idle, but I can change that to be the same as whatever is decided for CAN.

Reply:
resolved with latest commit